tools/building.py

```diff
@@ -438,8 +438,9 @@ def llvm_backend_args():
     allowed = ','.join(Settings.EXCEPTION_CATCHING_ALLOWED or ['__fake'])
     args += ['-emscripten-cxx-exceptions-allowed=' + allowed]
 
-  # asm.js-style setjmp/longjmp handling
-  args += ['-enable-emscripten-sjlj']
+  if Settings.SUPPORT_LONGJMP:
+    # asm.js-style setjmp/longjmp handling
+    args += ['-enable-emscripten-sjlj']
```

Comment:
Doesn't this change mean that we would break projects that set `SUPPORT_LONGJMP` only at link time?   

Not sure that is great idea.  Also its nice to be able to make as many flags as possible into link-only flags.

Reply:
Oh.. SUPPORT_LONGJMP=0 is the default so maybe not an issue?

Reply:
Yes, there is some risk of breakage, fair point. But OTOH if the flag is ignored at compile time, then just setting the flag at link time won't work for purposes of invokes - the object file will have been rewritten to use invokes already.

Maybe it's not worth making this change as proper invoke handling is landing any time now, but it does seem more correct to me, that if the flag is passed, we should assume there is no longjmp work to be done.